### DISCUSSION TITLE
The service is deployed in the Docker container. The IP address of the master registered in the ZooKeeper is 172.18. 0.2.（Docker 容器中部署服务，zk 中注册的 master 信息 ip 地址是 172.18.0.2。）

### DISCUSSION THREAD

**Discussion by caolist** [apache/dolphinscheduler#9241] [Q&A]
When I deploy services in the Docker container, I find that the IP addresses registered by Master and Work in the ZooKeeper are Docker's built-in network address 172.18.0.2, which is also displayed on the service management page. In addition, the task (the simplest shell) cannot run properly. Suspicion has something to do with this.

Does anyone know how to solve this problem? Thank you.

**Comment by caolist** [comment]
If the DS has multiple NICs, select the first NIC. Therefore, select the IP address of the default network eth0 of the Docker container. What can I do?

**Reply by chengqiZ** [reply]
I encountered a similar problem when deploying Nacos in docker. I wonder if it will help you.
1. Config Docker network to host .
2. Set an explicit registration IP.
3. Configure the selection policy of registered network card
